katz/src/main/kotlin/katz/instances/EitherTMonad.kt

```diff
@@ -0,0 +1,29 @@
+package katz
+
+class EitherTMonad<F, L>(val MF : Monad<F>) : Monad<EitherTF<F, L>> {
+    override fun <A> pure(a: A): EitherT<F, L, A> =
+            EitherT(MF, MF.pure(Either.Right(a)))
+
+    override fun <A, B> map(fa: EitherTKind<F, L, A>, f: (A) -> B): EitherT<F, L, B> =
+            fa.ev().map { f(it) }
+
+    override fun <A, B> flatMap(fa: EitherTKind<F, L, A>, f: (A) -> EitherTKind<F, L, B>): EitherT<F, L, B> =
+            fa.ev().flatMap { f(it).ev() }
+
+    override fun <A, B> tailRecM(a: A, f: (A) -> HK<EitherTF<F, L>, Either<A, B>>): EitherT<F, L, B> =
```

Comment:
i think we should test at least this method for every monad in order to avoid stack overflows